configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/PatientDetailsViewModel.kt

```diff
@@ -0,0 +1,358 @@
+      .take(MAX_RESOURCE_COUNT)
+      .map { createConditionItem(it, getApplication<Application>().resources) }
+      .let { conditions.addAll(it) }
+    return conditions
+  }
+
+  private suspend fun getPatientDetailDataModel(): List<PatientDetailData> {
+    val data = mutableListOf<PatientDetailData>()
+    val patient = getPatient()
+    patient.riskItem = getPatientRiskAssessment()
+
+    val observations = getPatientObservations()
+    val conditions = getPatientConditions()
+
+    patient.let { patientItem ->
+      data.add(PatientDetailOverview(patientItem, firstInGroup = true))
+      data.add(
+        PatientDetailProperty(
+          PatientProperty(getString(R.string.patient_property_id), patientItem.resourceId)
+        )
+      )
+      data.add(
+        PatientDetailProperty(
+          PatientProperty(
+            getString(R.string.patient_property_dob),
+            patientItem.dob?.localizedString ?: ""
+          )
+        )
+      )
+      data.add(
+        PatientDetailProperty(
+          PatientProperty(
+            getString(R.string.patient_property_gender),
+            patientItem.gender.replaceFirstChar {
+              if (it.isLowerCase()) it.titlecase(Locale.ROOT) else it.toString()
+            }
+          ),
+          lastInGroup = true
+        )
+      )
+    }
+
+    if (observations.isNotEmpty()) {
+      data.add(PatientDetailHeader(getString(R.string.header_observation)))
+
+      val observationDataModel =
+        observations.mapIndexed { index, observationItem ->
+          PatientDetailObservation(
+            observationItem,
+            firstInGroup = index == 0,
+            lastInGroup = index == observations.size - 1
+          )
+        }
+      data.addAll(observationDataModel)
+    }
+
+    if (conditions.isNotEmpty()) {
+      data.add(PatientDetailHeader(getString(R.string.header_conditions)))
+      val conditionDataModel =
+        conditions.mapIndexed { index, conditionItem ->
+          PatientDetailCondition(
+            conditionItem,
+            firstInGroup = index == 0,
+            lastInGroup = index == conditions.size - 1
+          )
+        }
+      data.addAll(conditionDataModel)
+    }
+
+    return data
+  }
+
+  private val LocalDate.localizedString: String
+    get() {
+      val date = Date.from(atStartOfDay(ZoneId.systemDefault())?.toInstant())
+      return if (isAndroidIcuSupported())
+        DateFormat.getDateInstance(DateFormat.DEFAULT).format(date)
+      else SimpleDateFormat.getDateInstance(DateFormat.DEFAULT, Locale.getDefault()).format(date)
+    }
+
+  // Android ICU is supported API level 24 onwards.
+  private fun isAndroidIcuSupported() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
+
+  private fun getString(resId: Int) = getApplication<Application>().resources.getString(resId)
+
+  private suspend fun getPatientRiskAssessment(): RiskAssessmentItem {
```

Comment:
We aren't using any of the risk assessment functions so we should remove them

Reply:
Do you think we should have it?

Reply:
It's not being used at all. The demo app had logic to specifically determine risk and display risk status.

Reply:
Removed. Done.